common/src/main/java/org/opensearch/ml/common/model/TextEmbeddingModelConfig.java

```diff
@@ -263,4 +290,17 @@ public static FrameworkType from(String value) {
         }
     }
 
+    private void validateNoDuplicateKeys(String allConfig, Map<String, Object> additionalConfig) {
```

Comment:
You can use the same function in base class with changing `private` to `protected/public`. You don't have to hold a duplicated function for each child class.
Or putting this function in some util class is also OK, just don't have exactly same functions at multiple places.